### PR TITLE
Port OneLogin.Events to pantherlog and add username tag

### PR DESCRIPTION
**Pull request by alxarch**
## Background

Why are you making this change? Please provide a reference to any related issues and PRs

## Changes

- Port OneLogin to `pantherlog` engine
- Convert tests to YAML

## Testing

- mage test:go

### CODE REVIEW

**Review by giorgosp — APPROVED** [review]

File: internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go [hunk 1]
```diff
@@ -32,113 +29,78 @@ func LogTypes() logtypes.Group {
...
+	Resolution                      pantherlog.Int64  `json:"resolution" description:"The resolution."`
+	DirectoryID                     pantherlog.Int64  `json:"directory_id" description:"The directory ID."`
+	AuthenticationFactorID          pantherlog.Int64  `json:"authentication_factor_id" description:"The ID of the authentication factor used."`
+	RiskCookieID                    pantherlog.String `json:"risk_cookie_id" description:"The ID of the risk cookie."`
+	AppID                           pantherlog.Int64  `json:"app_id" description:"ID of the app involved in the event, if applicable."`
+	CustomMessage                   pantherlog.String `json:"custom_message" description:"More details about the event."`
+	BrowserFingerprint              pantherlog.String `json:"browser_fingerprint" description:"The fingerprint of the browser."`
+	OtpDeviceName                   pantherlog.String `json:"otp_device_name" description:"Name of a device involved in the event."`
+	ActorUserName                   pantherlog.String `json:"actor_user_name" description:"First and last name of the user whose action triggered the event."`
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go:71]: Include this in the list of usernames? `panther:"username"  `

File: internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go [hunk 2]
```diff
@@ -32,113 +29,78 @@ func LogTypes() logtypes.Group {
...
+	ActorSystem                     pantherlog.String `json:"actor_system" description:"Acting system that triggered the event when the actor is not a user."`
+	UserFieldName                   pantherlog.String `json:"user_field_name" description:"The name of the custom user field."`
+	UserFieldID                     pantherlog.String `json:"user_field_id" description:"The ID of the custom user field."`
+	AssumingActingUserID            pantherlog.Int64  `json:"assuming_acting_user_id" description:"ID of the user who assumed the role of the acting user to trigger the event, if applicable."`
+	APICredentialName               pantherlog.String `json:"api_credential_name" description:"The name of the API credential used."`
+	ImportedUserName                pantherlog.String `json:"imported_user_name" description:"The name of the imported user."`
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go:77]: Include this in the list of usernames? panther:"username"

**Review by kostaspap — APPROVED** [review]

File: internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go [hunk 3]
```diff
@@ -32,113 +29,78 @@ func LogTypes() logtypes.Group {
...
-func (p *OneLoginParser) LogType() string {
-	return TypeOneLogin
-}
-
-func (event *OneLogin) updatePantherFields(p *OneLoginParser) {
-	event.SetCoreFields(p.LogType(), (*timestamp.RFC3339)(event.EventTimestamp), event)
-	event.AppendAnyIPAddressPtr(event.IPAddr)
+	UUID                            pantherlog.String `json:"uuid" validate:"required,uuid" description:"The Universal Unique Identifier for this message generated by OneLogin."`
+	AccountID                       pantherlog.Int64  `json:"account_id" validate:"required" description:"Account that triggered the event."`
+	EventTimestamp                  pantherlog.Time   `json:"event_timestamp" event_time:"true" tcodec:"layout=2006-01-02 15:04:05 MST" validate:"required" description:"Time and date at which the event was created. This value is autogenerated by OneLogin."`
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go:47]: Seems you can delete the `timestamp.OneLoginTimestamp` struct?

File: internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go (hunk 2 above)
Comment (alxarch) [reply internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go:77]: I am not sure what imported user name means, might lead to collisions in detections.

File: internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go (hunk 3 above)
Comment (alxarch) [reply internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go:47]: Ah yes! forgot that

File: internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go (hunk 1 above)
Comment (alxarch) [reply internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go:71]: The description says first and last name, do we want to use such fields for 'username'? I don't thing free text first/last name is helpful as an indicator.

File: internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go (hunk 3 above)
Comment (alxarch) [reply internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go:47]: Fixed

File: internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go (hunk 1 above)
Comment (kostaspap) [reply internal/log_analysis/log_processor/parsers/oneloginlogs/onelogin.go:71]: `user_name`, `actor_user_name`, `imported_user_name` they are all in the same format (first/last, service name (e.g. GSuite),...). A name that appears in some events as `user_name` can be `actor_user_name` in a different event.

If we import one, we should also import the others.

**Pull request merged**